lib/Controller/PageController.php

```diff
@@ -268,6 +282,10 @@ public function index(string $token = '', string $callUser = '', string $passwor
 		$csp->addAllowedConnectDomain("'self'");
 		$csp->addAllowedImageDomain('https://*.tile.openstreetmap.org');
 		$response->setContentSecurityPolicy($csp);
+		if ($throttle) {
+			// Logged-in user tried to access a chat they can not access
+			$response->throttle();
```

Comment:
Is this needed? From the endpoints that end in `index` only the authenticate endpoint is annotated with `@BruteForceProtection`, but if a registered user tries to load a group conversation that she is not a member of she will be [redirected to the main page without even checking the password](https://github.com/nextcloud/spreed/blob/384d25896f9a26e539d265bf9c388076f4fe9661/lib/Controller/PageController.php#L201-L210). If the conversation is a public one and a password is required it will be verified, but that code path is already throttled.

Reply:
We should have it, otherwise logged in users can bruteforce their way to public conversations they are not a member of.
But you are right, the BruteForceProtection annotation is missing

Reply:
Ah, right, I was thinking only in terms of passwords, but tokens should be protected too, indeed.

Reply:
Maybe I am testing it wrong, but I am not able to get the throttling for this. But independently of that, is it needed (different reasoning than before, keep reading :-P )? No matter if the token is found or not found the response is the same, so even if the endpoint is not bruteforce protected it is not possible to know if the token is valid or not just from doing GET requests to `index.php/apps/spreed?token=XXX`.

But if I am missing something and the bruteforce protection is needed, then shouldn't `showCall` be protected as well (as it is the method called for `index.php/call/XXX`)?

Reply:
> But if I am missing something and the bruteforce protection is needed, then shouldn't showCall be protected as well (as it is the method called for index.php/call/XXX)?

Added

> Maybe I am testing it wrong, but I am not able to get the throttling for this.

For logged in users there is no content wise difference, but if it is an existing room you join it, otherwise no new room is being added. So yeah doesn't seem to be very necessary atm, but I prefer keeping it additionally?

For guests it makes the difference whether it shows the conversation or a redirect to the login page

Reply:
> For logged in users there is no content wise difference, but if it is an existing room you join it, otherwise no new room is being added. So yeah doesn't seem to be very necessary atm, but I prefer keeping it additionally?

Fine by me to keep it. But I still have not been able to make the throttling work for that endpoint as a registered user :shrug: I tried with `for i in {0..10}; do time curl --user {USER_ID}:{PASSWORD} "https://{NEXTCLOUD_SERVER}/index.php/apps/spreed?token=XXX" >/dev/null 2>&1; done`, but all requests take (roughly) the same time (and, as far as I can tell, `response->throttle()` is being executed). However, if the `--user {USER_ID}:{PASSWORD}` part is removed (so the requests are made as a guest, so different code path in same endpoint) the throttling works as expected.
 
> For guests it makes the difference whether it shows the conversation or a redirect to the login page

:+1:

Reply:
Maybe I forgot to amend something before pushing. Let me retry tomorrow

Reply:
The reason is https://github.com/nextcloud/server/pull/33153 which breaks on curl when using `--user` as you always pass the login and therefor no attempts are counted => no sleep.